rachellegelden/Home-Together-Canada
# HTC-322: added search filters to member registration

# [HTC-322](https://github.com/rachellegelden/Home-Together-Canada/issues/322)

## Summary
- added the search filters to the route to register a member
- made search filter fields mandatory
- added validation to fields

## Relevant Motivation & Context
Save user filters to their account

## Testing Instructions
Using Postman, you will have to send a request to create a new member. You can use the following `JSON` as the `body` of the request. Make sure that the user is created with the correct fields. Try to enter some invalid fields and make sure it is rejected.

```
{
    "username": "rachellegelden111111",
    "password": "password",
    "email": "rachelle111111@gelden.com",
    "firstName": "Rachelle",
    "lastName": "Gelden",
    "phoneNumber": 4033365156,
    "addressLine1": "1 Oak Ave",
    "city": "Kelowna",
    "province": "BC",
    "postalCode": "T1V1S4",
    "hasDifferentMailingAddress": false,
    "gender": "Female",
    "birthYear": 1998,
    "status": "Existing Group",
    "workingStatus": "Student",
    "minMonthlyBudget": 500,
    "maxMonthlyBudget": 800,
    "hasHomeToShare": false,
    "isReligionImportant": false,
    "isDietImportant": false,
    "hasHealthMobilityIssues": false,
    "hasAllergies": false,
    "hasPets": false,
    "isSmoker": false,
    "numRoommates": 1,
    "areasOfInterest": [{"province": "AB", "city": "Calgary", "radius": 80}],
    "minAgePreference": 18,
    "maxAgePreference": 25,
    "minBudgetPreference": 500,
    "maxBudgetPreference": 1000,
    "statusPreference": ["Single", "Couple"],
    "minNumRoommatesPreference": 1,
    "maxNumRoommatesPreference": 3,
    "dietPreference": false,
    "petsPreference": true,
    "smokingPreference": false,
    "healthAndMobilityPreference": false,
    "genderPreference": ["Female", "Male"],
    "religionPreference": true,
    "othersWithHomeToSharePreference": false
}
```

## ⚠️ IMPORTANT ⚠️ 
Users will not be able to register as members through the site until the front-end also includes the search filters

## Developer checklist prior to opening this pull request:


## Reviewer
